CONTRIBUTING.md

```diff
@@ -0,0 +1,27 @@
+# Contributor Guide
+
+## Setting up a KFServing Environment
+TODO flesh this out
+1. make all
+2. make docker-build
+3. make docker-push
+4. make deploy
+5. make install
+4. kubectl apply -f docs/samples/tensorflow.yaml
+
+## Updating Dependencies
+Go projects follow the pattern of committing vendor code to the repo. We will update dependencies frequently to solve upgrade conflicts early.
+
+1. dep ensure
+2. git commit -am "Updates vendor dependencies"
+
+## Submitting a PR
+The following should be viewed as Best Practices unless you know better ones (please submit a guidelines PR).
+
+| Practice | Rationale |
+| ----- | --------- |
+| Keep the code clean | The health of the codebase is imperative to the success of the project. Files should be under 500 lines long in most cases, which may mean a refactor is necessary before adding changes. |
+| Squash your commits | A clean codebase needs a clean commit log. Aim to submit one commit per PR, which may mean squashing your changes. You'll thank yourself later if you ever need to revert changes you've made and you'll simplify the lives of anyone looking into the history of the project. |
```

Comment:
Does PR merge process automatically squash commits in a PR when creating a merge commit?

Reply:
I belive it does @abhi-g

Reply:
oh awesome. I guess your commit messages just get squashed instead of fixup